tests/microbenchmarks/attention_perf/test_attention_perf_module.py

```
@@ -0,0 +1,263 @@
+    golden_entry = _golden_for(case.case_id, sig.arch)
+    if golden_entry is None:
+        _record(sig, None, verdict="bootstrap", observed_cv=round(cv, 4))
+        _bootstrap_continuous_msg(case.case_id, sig.arch, observed, cv, len(sig.raw_times_ms))
+
+    baseline, margin, threshold = _continuous_gate(golden_entry)
+    verdict = "pass" if observed <= threshold else "REGRESSION"
+    _record(sig, baseline, verdict=verdict,
+            observed_cv=round(cv, 4), threshold_ms=round(threshold, 4))
+    assert observed <= threshold, (
+        f"{what} gpu_time regression on {sig.arch}: observed={observed:.4f}ms > "
+        f"threshold={threshold:.4f}ms (baseline={baseline:.4f}ms × {1 + margin:.4f}; "
```

Ruff (0.15.18)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)